django/db/models/base.py

```diff
@@ -1052,7 +1052,7 @@ def _save_table(
                 )
             fields = meta.local_concrete_fields
             if not pk_set:
-                fields = [f for f in fields if f is not meta.auto_field]
+                fields = [f for f in fields if f not in meta.auto_fields]
```

Comment:
I assume that it's the only bit of logic that is actually preventing an out-of-core implementation?

Maybe we could actually swap this logic to use `field.db_returning` instead?